tests/unit/cluster/failover2.tcl

```diff
@@ -0,0 +1,62 @@
+# Check the basic monitoring and failover capabilities.
+
+start_cluster 3 6 {tags {external:skip cluster} overrides {cluster-ping-interval 1000 cluster-node-timeout 15000}} {
+
+test "Cluster is up" {
+    wait_for_cluster_state ok
+}
+
+test "Cluster is writable" {
+    cluster_write_test [srv 0 port]
+}
+
+set paused_pid [srv 0 pid]
+test "Killing one primary node" {
+    pause_process $paused_pid
+}
+
+test "Wait for failover" {
+    wait_for_condition 1000 50 {
+        [s -3 role] == "master" || [s -6 role] == "master"
+    } else {
+        fail "No failover detected"
+    }
+}
+
+test "Killing the new primary node" {
+    if {[s -3 role] == "master"} {
+        set replica_to_be_primary -6
+        set paused_pid2 [srv -3 pid]
+    } else {
+        set replica_to_be_primary -3
+        set paused_pid2 [srv -6 pid]
+    }
+    pause_process $paused_pid2
+}
+
+test "Cluster should eventually be up again" {
+    for {set j 0} {$j < [llength $::servers]} {incr j} {
+        if {[process_is_paused $paused_pid]} continue
+        if {[process_is_paused $paused_pid2]} continue
```

Comment:
One of these two should always be true for all values of j, so won't this skip it for everything? I think you're missing some check for j in here to skip the paused nodes.

Reply:
right, i copy the logic from other tests, and now i see in #442, i've made this mistake already...

Reply:
#783